zerver/lib/cache.py

```diff
@@ -534,3 +534,33 @@ def wrapper(*args: Any, **kwargs: Any) -> Any:
         return wrapper
 
     return decorator
+
+def dict_to_items_tuple(user_function: Callable[..., Any]) -> Callable[..., Any]:
+    """Wrapper that converts any dict args to dict item tuples."""
+    def dict_to_tuple(k: Any) -> Any:
+        if isinstance(k, dict):
+            return tuple(k.items())
+        return k
+
+    def wrapper(*args: Any, **kwargs: Any) -> Any:
+        new_args = (dict_to_tuple(arg) for arg in args)
+        return user_function(*new_args, **kwargs)
+
+    return wrapper
+
+def items_tuple_to_dict(user_function: Callable[..., Any]) -> Callable[..., Any]:
+    """Wrapper that converts any dict items tuple args to dicts."""
+    def dict_items_to_dict(k: Any) -> Any:
+        if isinstance(k, tuple):
+            try:
+                return dict(k)
+            except TypeError:
+                pass
+        return k
```

Comment:
What's the use case for this `return k` case?

Reply:
We want to convert only dict item tuples to dicts, and all other kinds of arguments should be returned as they were passed.